gen/enum_test.go

```diff
@@ -501,6 +502,69 @@ func TestEnumLabelValid(t *testing.T) {
 	}
 }
 
+func TestEnumStrictMarshalText(t *testing.T) {
+	tests := []struct {
+		testName   string
+		enumStrict tems.EnumMarshalStrict
+		hasError   bool
+	}{
+		{"marshalText success",
+			1,
+			false,
+		},
+		{
+			"marshalText unknown value",
+			5,
+			true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(string(tt.testName), func(t *testing.T) {
+			marshalledValue, err := tt.enumStrict.MarshalText()
+			if tt.hasError {
+				assert.NotNil(t, err)
+				assert.Equal(t, fmt.Sprintf("unknown enum value %q for %q", tt.enumStrict, "EnumMarshalStrict"), err.Error())
+			} else {
+				assert.Nil(t, err)
+				assert.Equal(t, []byte(tt.enumStrict.String()), marshalledValue)
+			}
+		})
+	}
+}
+
+func TestEnumStrictMarshalJSON(t *testing.T) {
```

Comment:
This entire test case should also look roughly like the MarshalText test case. We might even be able to merge the two by having `wantText` and `wantJSON` instead of just `want` above.